tests/sentry/workflow_engine/endpoints/test_organization_detector_index.py

```diff
@@ -833,6 +883,281 @@ def test_owner_not_in_organization(self) -> None:
+            role="manager",
+            organization=self.organization,
+        )
+
+    def test_update_detectors_by_ids_success(self) -> None:
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params=[("id", str(self.detector.id)), ("id", str(self.detector_two.id))],
+            enabled=False,
+            status_code=200,
+        )
+
+        # Verify detectors were updated
+        self.detector.refresh_from_db()
+        self.detector_two.refresh_from_db()
+        assert self.detector.enabled is False
+        assert self.detector_two.enabled is False
+
+        # Verify third detector was not affected
+        self.detector_three.refresh_from_db()
+        assert self.detector_three.enabled is True
+
+        # Verify response data
+        assert len(response.data) == 2
+        detector_ids = {d["id"] for d in response.data}
+        assert detector_ids == {str(self.detector.id), str(self.detector_two.id)}
+        assert all(d["enabled"] is False for d in response.data)
+
+    def test_update_detectors_by_query_success(self) -> None:
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"query": "test", "project": self.project.id},
+            enabled=False,
+            status_code=200,
+        )
+
+        # Verify detector matching query was updated
+        self.detector.refresh_from_db()
+        assert self.detector.enabled is False
+
+        # Verify other detectors were not affected
+        self.detector_two.refresh_from_db()
+        self.detector_three.refresh_from_db()
+        assert self.detector_two.enabled is True
+        assert self.detector_three.enabled is True
+
+        # Verify response
+        assert len(response.data) == 1
+        assert response.data[0]["id"] == str(self.detector.id)
+        assert response.data[0]["enabled"] is False
+
+    def test_update_detectors_enable_success(self) -> None:
+        self.detector.update(enabled=False)
+        self.detector_two.update(enabled=False)
+        self.detector_three.update(enabled=False)
+
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"id": str(self.detector_three.id)},
+            enabled=True,
+            status_code=200,
+        )
+
+        # Verify detector was enabled
+        self.detector_three.refresh_from_db()
+        assert self.detector_three.enabled is True
+
+        # Verify response
+        assert len(response.data) == 1
+        assert response.data[0]["id"] == str(self.detector_three.id)
+        assert response.data[0]["enabled"] is True
+
+    def test_update_detectors_no_parameters_error(self) -> None:
+        response = self.get_error_response(
+            self.organization.slug,
+            enabled=False,
+            status_code=400,
+        )
+
+        assert "At least one of 'id', 'query', 'project', or 'projectSlug' must be provided" in str(
```

Comment:
I don't see tests for if you pass `projectSlug` or `project` on it's own without an additional qs - is the expected behavior that all detectors in the matching project(s) will be updated?

Reply:
Since the filtering logic is shared by GET/PUT/DELETE I didn't test all filtering for every endpoint, but here's an example from the GET endpoint tests that passes `project` without any other queries

https://github.com/getsentry/sentry/blob/22c634be89f1c3efa3b626cfb4031a13e36bf5ca/tests/sentry/workflow_engine/endpoints/test_organization_detector_index.py#L57-L67

If only a `query` is passed, then all matching detectors in projects that the user is a member of will be updated. If only `project` is passed, then all detectors owned by the project will be updated.